### PR TITLE
Display avatar in video miniatures

### LINKED ISSUE #4560: Display avatar in video miniatures
**Describe the problem to be solved**
It's not obvious for a user to detect the video author in video miniatures and video watch page since the author name is only written in text.

**Describe the solution you would like:**
Possibility to display avatar in video miniature and video watch page.

**Describe alternatives you have considered**  
The current avatar images are 120x120 px, whilst an avatar in the video miniature would be around 35x35 px. I can see the following alternatives.

A) CSS resize current avatar images. This will mean bigger files than necessary. I tried to convert a 120x120 avatar to 35x35 and saw a reduction from 8,9 kb to 2,1 kb.
B) Store two different versions of avatars; one small and one big. This will mean some more storage, but compared to the video file sizes it's negligible. 10 000 users with custom avatars would mean 20 mb extra storage, based on my test above.

### PR DESCRIPTION
**Pull request by kontrollanten**
## Description
Display avatars in video miniatures

## Related issues
closes #4560

## Has this been tested?

- [x] 🙅 no, because this PR does not update server code

## Screenshots
image

### CODE REVIEW

**Comment by Chocobozzz** [comment]
Please use an opt-in config for now: https://github.com/Chocobozzz/PeerTube/blob/develop/config/default.yaml#L624

(thanks for the videos list fix, I put it in develop and added a test in https://github.com/Chocobozzz/PeerTube/commit/242f52253e46ce0ffd7349cb06bcd887bb89cf06)

**Comment by kontrollanten** [comment]
Fixed @Chocobozzz It was pretty hard find a nice way to set component input default value from the config. The other option I found was to set the value in every places that my-videos-list is used, but that wouldn't be totally nice either.

File: config/default.yaml [hunk 1]
```diff
@@ -624,6 +624,7 @@ client:
     miniature:
       # By default PeerTube client displays author username
       prefer_author_display_name: false
+      display_avatar: false
```
Comment (Chocobozzz) [inline config/default.yaml:627]: `display_author_avatar` for consistency

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

File: server/initializers/config.ts [hunk 2]
```diff
@@ -62,7 +62,8 @@ const CONFIG = {
   CLIENT: {
     VIDEOS: {
       MINIATURE: {
-        get PREFER_AUTHOR_DISPLAY_NAME () { return config.get<boolean>('client.videos.miniature.prefer_author_display_name') }
+        get PREFER_AUTHOR_DISPLAY_NAME () { return config.get<boolean>('client.videos.miniature.prefer_author_display_name') },
+        get DISPLAY_AVATAR () { return config.get<boolean>('client.videos.miniature.display_author_avatar') }
```
Comment (Chocobozzz) [inline server/initializers/config.ts:66]: `DISPLAY_AUTHOR_AVATAR`

File: server/lib/server-config-manager.ts [hunk 3]
```diff
@@ -46,6 +46,7 @@ class ServerConfigManager {
       client: {
         videos: {
           miniature: {
+            displayAvatar: CONFIG.CLIENT.VIDEOS.MINIATURE.DISPLAY_AVATAR,
```
Comment (Chocobozzz) [inline server/lib/server-config-manager.ts:49]: `displayAuthorAvatar`

File: shared/models/server/server-config.model.ts [hunk 4]
```diff
@@ -38,6 +38,7 @@ export interface ServerConfig {
   client: {
     videos: {
       miniature: {
+        displayAvatar: boolean
```
Comment (Chocobozzz) [inline shared/models/server/server-config.model.ts:41]: `displayAuthorAvatar`

File: client/src/app/shared/shared-video-miniature/videos-list.component.ts [hunk 5]
```diff
@@ -161,6 +172,14 @@ export class VideosListComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   ngOnChanges (changes: SimpleChanges) {
+    if (changes['displayOptions'] || !this.displayOptions) {
+      this.displayOptions = {
+        ...this.defaultDisplayOptions,
+        avatar: this.serverService.getHTMLConfig().client.videos.miniature.displayAvatar,
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-video-miniature/videos-list.component.ts:178]: `displayAuthorAvatar`

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

**Pull request merged**